feat(auth): make login otp email informational

LinHuiqing opened the pull request:
- add more general information i.e. where email is coming from
- add IP address where login request is coming from

closes #39

## Problem

Closes #39.

## Solution

**Features**:

- add client machine's IP address information which sent tried to login

**Improvements**:

- add information on what OTP is for
- add fallback instructions if OTP does not work

## Before & After Screenshots

**BEFORE**:
<img width="228" alt="Screenshot 2021-09-07 at 4 44 11 PM" src="https://user-images.githubusercontent.com/37061143/132314127-50955f7b-4923-4036-a533-1bf1370511f2.png">

**AFTER**:
<img width="1014" alt="Screenshot 2021-09-07 at 8 55 59 PM" src="https://user-images.githubusercontent.com/37061143/132428005-d62f44e5-bca3-4ace-84f4-6416c1deaf3c.png">

## Tests

- (In test suite) `sendLoginOtp` in `src/modules/mail/__tests__/mail.service.spec.ts`.
- When user enters their email into the form (at https://ask.gov.sg/login), they should receive an email similar to what is seen in the after screenshot.
  - User should be able to login by entering the OTP in the login form.
  - User should be able to match the IP address to their own IP address, or find the rough location where the IP is from on the internet.

LinHuiqing committed:
feat(auth): make login otp email informational

- add more general information i.e. where email is coming from
- add IP address where login request
is coming from

closes #39

LinHuiqing committed:
fix(auth): modify url to direct user to for fallback to login url

LoneRifle commented:
Please have a look at https://github.com/opengovsg/checkfirst/blob/develop/src/server/utils/express.ts to figure out how we should be getting the IP of the calling client when the request is coming through Cloudflare

LinHuiqing committed:
fix(auth): change auth client ip to take from cloudflare connecting ip

get client ip address from `CF-Connecting-IP` field of request header instead of from
`x-forwarded-for` field or socket remote address

LoneRifle commented on server/src/modules/auth/auth.controller.ts:
```diff
@@ -111,10 +111,11 @@ export class AuthController {
 
     const otp = generateRandomDigits(OTP_LENGTH)
     const hashedOtp = await hashData(otp)
+    const ip = req.header('CF-Connecting-IP') || req.ip
```
Maybe leave commentary here explaining what this is doing, and why. Take a cue from the comments in the code from where you found this line.

LoneRifle replied:
So ie...
```suggestion
    // On staging and production, we use CloudFlare which adds a CF-Connecting-IP
    // header with every request which contains only the origin IP
    // See https://support.cloudflare.com/hc/en-us/articles/200170986-How-does-CloudFlare-handle-HTTP-Request-headers-
    // As a fallback, use req.ip, automatically parsed from the x-forwarded-for
    // header if we configure app.set('trust proxy').
    const ip = req.header('CF-Connecting-IP') || req.ip
```

LoneRifle reviewed (approved):
lgtm otherwise
